quarkus/dist/src/main/content/bin/kc.bat

```diff
@@ -50,22 +48,31 @@ if "%KEY%" == "start-dev" (
   shift
   goto READ-ARGS
 )
-if not "%KEY:~0,2%"=="--" if "%KEY:~0,2%"=="-D" (
-  set SERVER_OPTS=%SERVER_OPTS% %KEY%=%2
+set "VALUE=%~2"
+set PROBABLY_VALUE=false
+if "%VALUE%" NEQ "" if "%VALUE:~0,1%" NEQ "-" if "%KEY:^==%"=="%KEY%" (
+  set PROBABLY_VALUE=true
+)
```

Comment:
I think I found it, Batch scripts can be sensitive to syntax, especially with nested conditions, it's sometimes difficult to spot, since syntactically there's nothing particularly wrong with that (especially when even wine can interpret it correctly) Anyway the following change should solve the issue.
```suggestion
if "%VALUE%" NEQ "" (
    if "%VALUE:~0,1%" NEQ "-" (
        if "%KEY:^==%"=="%KEY%" (
            set PROBABLY_VALUE=true
        )
    )
)
```

Reply:
Results with the proposed applied:
![image](https://github.com/user-attachments/assets/10867692-edb9-42c0-a4a4-3abd4754ad95)
(W11 23H2)